theory/DD/countpairs_impl.c.src

```diff
@@ -223,20 +223,44 @@ int countpairs_DOUBLE(const int64_t ND1, DOUBLE *X1, DOUBLE *Y1, DOUBLE *Z1,
         }
     }
 
-    if(options->boxsize < 0 && options->periodic){
-        fprintf(stderr, "boxsize = %g must be >= 0 with periodic wrap. Please specify a non-zero boxsize or zero to detect the particle extent.\n",
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wfloat-equal"
+#endif
+
+    if(options->boxsize == BOXSIZE_NOTGIVEN && options->periodic){
+        fprintf(stderr, "boxsize = %g must be specified with periodic wrap. Please specify a non-zero boxsize, or zero to detect the particle extent, or -1 to make a dimension non-periodic.\n",
             options->boxsize);
         return EXIT_FAILURE;
     }
-    const DOUBLE xdiff = options->boxsize > 0 ? options->boxsize:(xmax-xmin);
-    const DOUBLE ydiff = options->boxsize > 0 ? options->boxsize:(ymax-ymin);
-    const DOUBLE zdiff = options->boxsize > 0 ? options->boxsize:(zmax-zmin);
+    
+    const double boxsize_y = options->boxsize_y == BOXSIZE_NOTGIVEN ? options->boxsize : options->boxsize_y;
+    const double boxsize_z = options->boxsize_z == BOXSIZE_NOTGIVEN ? options->boxsize : options->boxsize_z;
+
+#ifdef __GNUC__
+#pragma GCC diagnostic pop
+#endif
+
+    const int periodic_x = options->periodic && options->boxsize_x >= 0;
+    const int periodic_y = options->periodic && boxsize_y >= 0;
+    const int periodic_z = options->periodic && boxsize_z >= 0;
+
+    const DOUBLE xdiff = (options->periodic && options->boxsize_x > 0) ? options->boxsize_x : (xmax-xmin);
+    const DOUBLE ydiff = (options->periodic && boxsize_y > 0) ? boxsize_y : (ymax-ymin);
+    const DOUBLE zdiff = (options->periodic && boxsize_z > 0) ? boxsize_z : (zmax-zmin);
```

Comment:
Not sure I follow the logic here. It also seems that we might want to initialise to 0.0 for the cases where -1 has been passed as the box size. 

Separately, this might be a good opportunity to clean up the naming for x/y/zdiff -> x/y/z_wrap or something like that

Reply:
I've renamed these to `xwrap` now. These values are never used if a dimension is not periodic. We could set them to 0 in that case, but I'm not sure how to do that without adding another conditional, which I'm afraid would make the logic even harder to follow.